Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/cut_with_plane.h

```diff
@@ -0,0 +1,453 @@
+ *     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%edge_descriptor`
+ *                    as key type and `bool` as value type}
+ *     \cgalParamDefault{a constant property map returning `false` for any edge}
+ *   \cgalParamNEnd
+ *
+ *   \cgalParamNBegin{edge_is_marked_map}
+ *     \cgalParamDescription{a property map filled by this function that puts `true` for all intersection edge of faces
+ *                           of `pm` and `plane`, and `false` for all other edges.}
+ *     \cgalParamType{a class model of `WritablePropertyMap` with `boost::graph_traits<PolygonMesh>::%edge_descriptor`
+ *                    as key type and `bool` as value type}
+ *     \cgalParamDefault{a constant property map returning `false` for any edge}
+ *   \cgalParamNEnd
+ *
+ *   \cgalParamNBegin{vertex_oriented_side_map}
+ *     \cgalParamDescription{a property map filled by this function containing the position
+ *                           of each vertex relative to the oriented plane `plane`.}
+ *     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+ *                    as key type and `Oriented_side` as value type}
+ *     \cgalParamDefault{Dynamic vertex property map}
+ *     \cgalParamExtra{If the concurrenty tag is set to `Parallel_tag`, the property map might be filled by several thread at the same time.}
+ *   \cgalParamNEnd
+ *
+ *    \cgalParamNBegin{do_not_triangulate_faces}
+ *      \cgalParamDescription{If the input mesh is triangulated and this parameter is set to `false`, the mesh will be kept triangulated.}
+ *      \cgalParamType{`bool`}
+ *      \cgalParamDefault{`true`}
+ *    \cgalParamNEnd
+ *
+ *    \cgalParamNBegin{vertex_point_map}
+ *      \cgalParamDescription{a property map associating points to the vertices of `pm`}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+ *                     as key type and `GeomTraits::Point_3` as value type, `GeomTraits` being the type of the parameter `geom_traits`}
+ *      \cgalParamDefault{`boost::get(CGAL::vertex_point, pm)`}
+ *      \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t` must be available in `PolygonMesh`.}
+ *    \cgalParamNEnd
+ *
+ *    \cgalParamNBegin{geom_traits}
+ *      \cgalParamDescription{an instance of a geometric traits class}
+ *      \cgalParamType{a class model of `Kernel`}
+ *      \cgalParamDefault{a \cgal kernel deduced from the point type, using `CGAL::Kernel_traits`}
+ *      \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+ *    \cgalParamNEnd
+ *
+ *  \cgalNamedParamsEnd
+ */
+template <class PolygonMesh, class Plane_3, class NamedParameters =  parameters::Default_named_parameters>
+void cut_with_plane(PolygonMesh& pm,
```

Comment:
You do not cut, but you  refine.  So the function name is wrong. Cut sounds like clip

Reply:
`insert_intersection_with_plane()`?

Reply:
`refine()`  or `refine_with_plane()`.   The first verb even firrst word of the documentation is "refines".